tests/browser_context_test.go

```diff
@@ -913,3 +913,32 @@ func TestBrowserContextGrantPermissions(t *testing.T) {
 		})
 	}
 }
+
+func TestBrowserContextClearPermissions(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no_permissions_set", func(t *testing.T) {
+		t.Parallel()
+
+		tb := newTestBrowser(t)
+		bCtx, err := tb.NewContext(nil)
+		require.NoError(t, err)
+
+		err = bCtx.ClearPermissions()
```

Comment:
Should we first assert that permissions are not there?

Reply:
One way of doing this is actually doing the action that requires permission.

Reply:
Nice idea. I tried it with a few and I was unable to get them to work with the test browser. It works perfectly fine with headful browser test though 🤷 It could be an issue with headless browsers... not sure tbh.

Reply:
Hmm, never mind then :)

Reply:
Here's the diff of what one of the tests I tried:

<details>
  <summary>Click me</summary>

```diff
diff --git a/tests/browser_context_test.go b/tests/browser_context_test.go
index 8f1a351..b1a8681 100644
--- a/tests/browser_context_test.go
+++ b/tests/browser_context_test.go
@@ -848,25 +848,52 @@ func TestBrowserContextClearPermissions(t *testing.T) {
        t.Run("no_permissions_set", func(t *testing.T) {
                t.Parallel()
 
-               tb := newTestBrowser(t)
+               tb := newTestBrowser(t, withFileServer())
                bCtx, err := tb.NewContext(nil)
                require.NoError(t, err)
 
+               page, err := bCtx.NewPage()
+               require.NoError(t, err)
+
+               _, err = page.Goto(tb.staticURL("geolocation.html"), nil)
+               require.NoError(t, err)
+               text := page.Locator(`div[id="geo"]`, nil).InnerText(nil)
+               assert.Equal(t, "None", text)
+
                err = bCtx.ClearPermissions()
                assert.NoError(t, err)
+
+               _, err = page.Goto(tb.staticURL("geolocation.html"), nil)
+               require.NoError(t, err)
+               text = page.Locator(`div[id="geo"]`, nil).InnerText(nil)
+               assert.Equal(t, "None", text)
        })
 
        t.Run("permissions_set", func(t *testing.T) {
                t.Parallel()
 
-               tb := newTestBrowser(t)
+               tb := newTestBrowser(t, withFileServer())
                bCtx, err := tb.NewContext(nil)
                require.NoError(t, err)
 
                err = bCtx.GrantPermissions([]string{"geolocation"}, common.NewGrantPermissionsOptions())
                require.NoError(t, err)
 
+               page, err := bCtx.NewPage()
+               require.NoError(t, err)
+
+               _, err = page.Goto(tb.staticURL("geolocation.html"), nil)
+               require.NoError(t, err)
+               text := page.Locator(`div[id="geo"]`, nil).InnerText(nil)
+               assert.Contains(t, text, "Lat")
+               assert.Contains(t, text, "Long")
+
                err = bCtx.ClearPermissions()
                assert.NoError(t, err)
+
+               _, err = page.Goto(tb.staticURL("geolocation.html"), nil)
+               require.NoError(t, err)
+               text = page.Locator(`div[id="geo"]`, nil).InnerText(nil)
+               assert.Equal(t, "None", text)
        })
 }
```

```html
<html lang="en">
    <head></head>
    <body>
        <div id='geo' style='display: none;'>None</div>
        <script>
            function getGeolocation() {
                navigator.geolocation.getCurrentPosition(function (position) {
                    let lat = position.coords.latitude;
                    let long = position.coords.longitude;

                    document.getElementById("geo").innerHTML = "Lat: " + lat.toFixed(2) + " Long: " + long.toFixed(2) + "";
                });
            }
        </script>
    </body>
</html>

```

</details>

Reply:
This should work:

```diff
diff --git a/tests/browser_context_test.go b/tests/browser_context_test.go
index 8f1a351..13d7a8f 100644
--- a/tests/browser_context_test.go
+++ b/tests/browser_context_test.go
@@ -845,15 +845,32 @@ func TestBrowserContextGrantPermissions(t *testing.T) {
 func TestBrowserContextClearPermissions(t *testing.T) {
 	t.Parallel()
 
+	hasPermission := func(tb *testBrowser, p *common.Page, perm string) bool {
+		t.Helper()
+
+		js := fmt.Sprintf(`
+			(perm) => navigator.permissions.query(
+				{ name: %q }
+			).then(result => result.state)
+		`, perm)
+		v := p.Evaluate(tb.toGojaValue(js))
+
+		return tb.asGojaValue(v).String() == "granted"
+	}
+
 	t.Run("no_permissions_set", func(t *testing.T) {
 		t.Parallel()
 
 		tb := newTestBrowser(t)
 		bCtx, err := tb.NewContext(nil)
 		require.NoError(t, err)
+		p, err := bCtx.NewPage()
+		require.NoError(t, err)
+		require.False(t, hasPermission(tb, p, "geolocation"))
 
 		err = bCtx.ClearPermissions()
 		assert.NoError(t, err)
+		require.False(t, hasPermission(tb, p, "geolocation"))
 	})
 
 	t.Run("permissions_set", func(t *testing.T) {
@@ -862,11 +879,17 @@ func TestBrowserContextClearPermissions(t *testing.T) {
 		tb := newTestBrowser(t)
 		bCtx, err := tb.NewContext(nil)
 		require.NoError(t, err)
+		p, err := bCtx.NewPage()
+		require.NoError(t, err)
+
+		require.False(t, hasPermission(tb, p, "geolocation"))
 
 		err = bCtx.GrantPermissions([]string{"geolocation"}, common.NewGrantPermissionsOptions())
 		require.NoError(t, err)
+		require.True(t, hasPermission(tb, p, "geolocation"))
 
 		err = bCtx.ClearPermissions()
 		assert.NoError(t, err)
+		require.False(t, hasPermission(tb, p, "geolocation"))
 	})
 }
```